dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -21,21 +26,42 @@ namespace winrt::Microsoft
 
 namespace winrt::Microsoft::Windows::PushNotifications::implementation
 {
-    PushNotificationChannel::PushNotificationChannel(winrt::Windows::PushNotificationChannel const& channel): m_channel(channel) {}
-
     winrt::Windows::Uri PushNotificationChannel::Uri()
     {
-        return winrt::Windows::Uri{ m_channel.Uri() };
+        if (m_channel)
+        {
+            return winrt::Windows::Uri{ m_channel.Uri() };
+        }
+        else
+        {
+            return winrt::Windows::Uri{ m_channelInfo.channelUri };
+        }
     }
+
     winrt::Windows::DateTime PushNotificationChannel::ExpirationTime()
     {
-        return m_channel.ExpirationTime();
+        if (m_channel)
+        {
+            return m_channel.ExpirationTime();
+        }
+        else
+        {
+            return m_channelInfo.channelExpiryTime;
+        }
     }
+
     void PushNotificationChannel::Close()
     {
         try
         {
-            m_channel.Close();
+            if (m_channel)
+            {
+                m_channel.Close();
+            }
+            else
+            {
+               PushNotifications_CloseChannel(m_channelInfo.appUserModelId.c_str(), m_channelInfo.channelId.c_str());
```

Comment:
LOG_IF_FAILED ?

Reply:
Now TROW_IF_FAILED, so the error get cauth and looged into telemetry. That gives us a unified path for error handling.